server/routerlicious/packages/routerlicious-base/src/test/nexus/io.spec.ts

```diff
@@ -402,130 +413,601 @@ describe("Routerlicious", () => {
 				});
 
 				describe("#submitSignal", () => {
-					it("Single Signal", async () => {
-						const signalContent = "TestSignal";
-
-						const firstSocket = webSocketServer.createConnection();
+					const createSignalPromise = (client: Client): Promise<ISignalMessage> => {
+						return new Promise<ISignalMessage>((resolve) => {
+							const signalListener = (signal: ISignalMessage) => {
+								// Ignore disconnected signals
+								if (signal.clientId !== null) {
+									client.signalCount++;
+								}
+								client.socket.off("signal", signalListener);
+								resolve(signal);
+							};
+
+							client.socket.on("signal", signalListener);
+
+							// Timeout of 100ms to handle cases where no signal is received
+							setTimeout(() => {
+								client.socket.off("signal", signalListener);
+								resolve(undefined);
+							}, 100);
+						});
+					};
 
-						const firstConnectMessage = await connectToServer(
+					const createClient = async (
+						testId: string,
+						testTenantId: string,
+						testSecret: string,
+						v2: boolean = false,
+					): Promise<Client> => {
+						const socket = webSocketServer.createConnection();
+						const connectMessage = await connectToServer(
 							testId,
 							testTenantId,
 							testSecret,
-							firstSocket,
+							socket,
+							v2,
 						);
+						const clientId = connectMessage.clientId;
+						const client: Client = {
+							socket,
+							clientId,
+							signalCount: 0,
+						};
 
-						const secondSocket = webSocketServer.createConnection();
+						return client;
+					};
+
+					const stringSignalContent = "TestSignal";
+
+					let clients: Client[];
+
+					const numberOfClients = 3; // Change the amount of clients to test with (at leat 2 required)
 
-						await connectToServer(testId, testTenantId, testSecret, secondSocket);
+					assert(numberOfClients > 1, "Test requires at least 2 clients");
+
+					describe("v1 signals", () => {
+						beforeEach(async () => {
+							clients = [];
+							for (let i = 0; i < numberOfClients; i++) {
+								const client = await createClient(testId, testTenantId, testSecret);
+								clients.push(client);
+							}
+							clients.forEach((client) => {
+								client.signalPromise = createSignalPromise(client);
+							});
+						});
+						it("can handle a single broadcast signal", async () => {
+							clients[0].socket.send("submitSignal", clients[0].clientId, [
+								stringSignalContent,
+							]);
+
+							const userSignals = await Promise.all(
+								clients.map((client) => client.signalPromise),
+							);
+
+							clients.forEach((client, index) => {
+								assert.equal(
+									client.signalCount,
+									1,
+									`User ${index + 1} should have received 1 signal`,
+								);
+								assert.equal(
```

Comment:
`assert.equal` is not an appropriate test generally and is "lucky" here. We should use value comparisons rather than "reference" comparison which `assert.equal` is. For example: `assert.equal({}, {}, "will always fail);`. We use `deepEqual` later we should just use that here too.
Code deduplication is also very nice. We can have a helper for these common cases and make it much easier to see what the meaningful parts of the tests are.